pubsub/gcloud/aio/pubsub/subscription.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import logging
+
+import google.auth.exceptions
+import google.cloud.pubsub as pubsub
+import google.gax.errors
+import grpc
+
+
+log = logging.getLogger(__name__)
+
+
+class Subscription(pubsub.subscription.Subscription):
+    async def acknowledge(self, ack_ids, client=None, retries=0):
+        try:
+            return super().acknowledge(ack_ids, client=client)
+        except (google.auth.exceptions.TransportError,
+                google.gax.errors.RetryError):
+            # common intermittent communication errors
+            if not retries:
+                raise
+
+            await asyncio.sleep(1)
+        except Exception:
+            raise
+
+        return await self.acknowledge(ack_ids, client=client,
```

Comment:
Should we avoid recursion here and opt for more informative exceptions? This could make a pretty hard-to-read stacktrace. Otherwise, looks great!